core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -303,31 +304,31 @@ public void run() {
         return;
       }
 
-      Attributes.Builder attrs = Attributes.newBuilder();
+      ResolutionResult.Builder resultBuilder = ResolutionResult.newBuilder().setAddresses(servers);
       if (!resolutionResults.txtRecords.isEmpty()) {
         ConfigOrError serviceConfig =
             parseServiceConfig(resolutionResults.txtRecords, random, getLocalHostname());
         if (serviceConfig != null) {
-          if (serviceConfig.getError() != null) {
-            savedListener.onError(serviceConfig.getError());
-            return;
-          } else {
+          resultBuilder.setServiceConfig(serviceConfig);
+          if (serviceConfig.getError() == null) {
             @SuppressWarnings("unchecked")
             Map<String, ?> config = (Map<String, ?>) serviceConfig.getConfig();
```

Comment:
This doesn't seem right... You have changed `parseServiceConfig` to return the `ConfigOrError` returned from `serviceConfigParser`, which is no longer the raw format. To keep the old behavior that `GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG` is the raw config, you probably want to keep `parseServiceConfig()` as is, and instead call `serviceConfigParser` from here.